packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -161,6 +161,205 @@ void main() {
     expect(renderBox.size.width, 80.0);
   });
 
+  testWidgets('Leading and trailing spacing is correct with 0~2 destinations', (WidgetTester tester) async {
+    late StateSetter stateSetter;
+    List<NavigationRailDestination> destinations = const <NavigationRailDestination>[];
+    Widget? leadingWidget;
+    Widget? trailingWidget;
+
+    const Key leadingWidgetKey = Key('leadingWidget');
+    const Key trailingWidgetKey = Key('trailingWidget');
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(useMaterial3: true),
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            stateSetter = setState;
+            return Scaffold(
+              body: Row(
+                children: <Widget>[
+                  NavigationRail(
+                    destinations: destinations,
+                    selectedIndex: null,
+                    leading: leadingWidget,
+                    trailing: trailingWidget,
+                  ),
+                  const Expanded(
+                    child: Text('body'),
+                  ),
+                ],
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // empty destinations and leading widget
+    stateSetter(() {
+      destinations = const <NavigationRailDestination>[];
+      leadingWidget = FloatingActionButton(
+        key: leadingWidgetKey,
+        onPressed: () { },
+      );
+      trailingWidget = null;
+    });
+    await tester.pumpAndSettle();
+    expect(tester.renderObject<RenderBox>(find.byKey(leadingWidgetKey)).localToGlobal(Offset.zero), const Offset(0, 8.0));
+
+    // one destination and leading widget
+    stateSetter(() {
+      destinations = const <NavigationRailDestination>[
+        NavigationRailDestination(
+          icon: Icon(Icons.favorite_border),
+          selectedIcon: Icon(Icons.favorite),
+          label: Text('Abc'),
+        ),
+      ];
+    });
+    await tester.pumpAndSettle();
+    expect(tester.renderObject<RenderBox>(find.byKey(leadingWidgetKey)).localToGlobal(Offset.zero), const Offset(12.0, 8.0));
```

Comment:
Would be good if we can check that the first destination is located correctly:)

Reply:
I have updated the tests. I hope this is as expected.
[638feb4](https://github.com/flutter/flutter/pull/152972/commits/638feb413fc41d6ece834f93f1c6f090fc2272f7)

Reply:
Almost! It might be a little hard to understand these numbers. Following existing tests would be better, like this one: https://github.com/flutter/flutter/blob/603707836056cad5eeffd135d160d89d54782f2e/packages/flutter/test/material/navigation_rail_test.dart#L260

To make the test look shorter, we can convert the expect() part as a method and just call it by passing `nextDestinationY` and `renderBox`:
```dart
    void matchExpect(RenderBox renderBox, double nextDestinationY) {
      expect(
        renderBox.localToGlobal(Offset.zero),
        equals(
          Offset(
            (destinationWidth - renderBox.size.width) / 2.0,
            nextDestinationY + (destinationHeight - renderBox.size.height) / 2.0,
          ),
        ),
      );
    }
```
so we can directly check `matchExpect(firstRenderBox, nextDestinationY)`